devtools/maven/src/test/java/io/quarkus/maven/it/DevMojoIT.java

```diff
@@ -383,6 +389,51 @@ public void testThatNewBeanAreDiscovered() throws IOException, MavenInvocationEx
                 .atMost(1, TimeUnit.MINUTES).until(() -> getHttpResponse("/app/hello").contains("foobarbaz"));
     }
 
+    @Test
+    public void testThatTheApplicationOutputIsProperlyFormatted() throws MavenInvocationException, IOException {
+        testDir = initProject("projects/classic", "projects/project-classic-run-colored-output");
+        running = new RunningInvoker(testDir, false);
+
+        MavenProcessInvocationResult result = running.execute(Arrays.asList("package", "-DskipTests"), Collections.emptyMap());
+        await().atMost(1, TimeUnit.MINUTES).until(() -> result.getProcess() != null && !result.getProcess().isAlive());
+        assertThat(running.log()).containsIgnoringCase("BUILD SUCCESS");
+
+        Path jar = testDir.toPath().toAbsolutePath().resolve(Paths.get("target/acme-1.0-SNAPSHOT-runner.jar"));
+        File output = new File(testDir, "target/output.log");
+        output.createNewFile();
+
+        List<String> commands = new ArrayList<>();
+        commands.add(JavaBinFinder.findBin());
+        commands.add("-jar");
+        commands.add(jar.toString());
+        ProcessBuilder processBuilder = new ProcessBuilder(commands.toArray(new String[0]));
+        processBuilder.redirectOutput(output);
+        processBuilder.redirectError(output);
+        Process process = processBuilder.start();
+
+        // Wait until we get "org.acme" to make sure that the server is launched
+        await()
+                .pollDelay(1, TimeUnit.SECONDS)
+                .atMost(1, TimeUnit.MINUTES).until(() -> getHttpResponse("/app/hello/package").contains("org.acme"));
+
+        process.destroy();
+
+        String infoLogLevel = "INFO";
+        List<String> logs = FileUtils.readLines(output, "UTF-8");
+
+        assertThat(logs.isEmpty()).isFalse();
+        String firstLogLine = logs.get(0);
+        assertThat(firstLogLine.contains(infoLogLevel)).isTrue();
+        String date = firstLogLine.split(infoLogLevel)[0].trim();
+
+        assertThat(date.matches("^\\d{4}-\\d{2}-\\d{2}\\s\\d{2}:\\d{2}:\\d{2},\\d{3}$")).isTrue();
+        String secondLogLine = logs.get(1);
+        assertThat(secondLogLine.contains("Installed features: [cdi, resteasy, undertow-websockets]")).isTrue();
```

Comment:
I don't like these tests.  If we change the log output in any way it will break.  Fragile tests are not better than no tests IMO.  I think it's better to examine the actual log manager to ensure that it was installed and configured correctly, which is far less fragile.

Reply:
@dmlloyd Okay. I hear your point. Are you referring to this specific line or the whole test? 

Otherwise, I was thinking of adding a test case close to `LoggingSetupTemplate`. This will test if the configurations are correctly passed by probing `handlers` and do the necessary configurations. WDYT?

Reply:
I'm referring to all the assertions starting at line 460.

Reply:
> If you want it to still be black-box-like without relying on output details, I'd recommend testing that `java.util.logging.LogManager#getLogManager()` is an instance of `org.jboss.logmanager.LogManager`, and then ensure that the root logger's handler list contains a `ConsoleHandler`, and that its formatter is a `PatternFormatter` with the expected pattern, etc.

Done. I have added two test cases using `@QuarkusUnitTest`on the test-extension project. I could not figure out a better place to put them, I am open to suggestions on that.

Reply:
@dmlloyd Not sure I totally agree with that tbh. We lost the ability to test the real output and it got broken numerous times, without the tests themselves being broken.

I agree that it's not perfect and I wouldn't insist if we haven't had a lot of variations of log format failures but we got:
- native image broken and the rest working (this is not tested by the current state of the PR)
- quarkus:dev broken and the rest working (this is not tested by the current state of the PR)
- java -jar broken and the rest working (this is not tested by the current state of the PR)

When I say the rest working: the output was usually OK in the tests but not when running this particular thing. So I'm pretty the issue wouldn't be caught by the test now included in the PR.

So I think the initial approach is what we should include with the improvement that I would define the log format in the configuration so that we can be sure it doesn't change.

If we don't want to test the log output itself, we should have the application test the log manager and output something depending on if it's correctly set or not. But we need to run the real application in the various scenarios above.

Does it make sense?

Reply:
> @dmlloyd Not sure I totally agree with that tbh. We lost the ability to test the real output and it got broken numerous times, without the tests themselves being broken.
>
> I agree that it's not perfect and I wouldn't insist if we haven't had a lot of variations of log format failures but we got:
> 
> * native image broken and the rest working (this is not tested by the current state of the PR)
> * quarkus:dev broken and the rest working (this is not tested by the current state of the PR)
> * java -jar broken and the rest working (this is not tested by the current state of the PR)

However, there have been zero cases where the problem is the actual output from a correctly configured log handler.

> When I say the rest working: the output was usually OK in the tests but not when running this particular thing. So I'm pretty the issue wouldn't be caught by the test now included in the PR.
> 
> So I think the initial approach is what we should include with the improvement that I would define the log format in the configuration so that we can be sure it doesn't change.
> 
> If we don't want to test the log output itself, we should have the application test the log manager and output something depending on if it's correctly set or not. But we need to run the real application in the various scenarios above.

Sure, I agree that the scenarios should be run, I am simply saying that we should not be testing the actual text output because that'd be just another bad test to maintain.

Reply:
I have pushed two commits one adding tests as par @gsmet proposition touching
- quarkus:dev app execution
- java -jar broken app execution

Also I have kept the @dmlloyd suggestions of adding logger configuration tests.

WDYT?